packages/node_modules/@node-red/nodes/core/function/10-function.js

```diff
@@ -16,6 +16,19 @@
 
 module.exports = function(RED) {
     "use strict";
+
+    function LibsConfigNode(n) {
+        RED.nodes.createNode(this, n);
+        this.name = n.name;
+        this.libs = n.libs;
+    }
+    RED.nodes.registerType("library-config", LibsConfigNode);
+
+    RED.httpNode.get("/function/modules", function (req, res) {
```

Comment:
All admin related routes must be on `RED.httpAdmin` and use the `needsPermission()` middleware to ensure it can be secured.

Reply:
See my other comment around why `listNPMModules` is exposed to nodes: if this functionality is core node-red functionality, then this admin route does not belong inside the Function node's code and should be exposed as a core admin api.

Reply:
As discussed here: https://github.com/node-red/designs/pull/31#discussion_r545746805

Reply:
Thanks for your comment. I will consider moving it to core admin API.